core/src/main/java/com/linecorp/armeria/server/HttpResponseSubscriber.java

```diff
@@ -232,6 +235,11 @@ public void onError(Throwable cause) {
             failAndRespond(cause,
                            AggregatedHttpMessage.of(((HttpStatusException) cause).httpStatus()),
                            Http2Error.CANCEL);
+        } else if (cause instanceof AbortedStreamException) {
+            // One of the two cases:
+            // - Client closed the connection too early.
+            // - Response publisher aborted the stream.
+            failAndReset((AbortedStreamException) cause);
```

Comment:
/cc @kojilin You may want to send a canary to make sure this PR fixes your problem.